om4labs/om4common.py

```diff
@@ -462,10 +483,26 @@ def read_topography(dictArgs):
         ds = cat["topog"].to_dask()
 
     if "deptho" in list(ds.variables):
-        depth = ds.deptho.to_masked_array()
+        depth = ds.deptho
     elif "depth" in list(ds.variables):
-        depth = ds.depth.to_masked_array()
+        depth = ds.depth
+
+    if outputgrid == "symetric":
+        """ extend array, assuming periodic/tripolar grid"""
+        warnings.warn("assuming the grid is periodic and tripolar")
+        tmp = depth.values
+        inner = xr.DataArray(tmp, dims=("y", "x"))
+
+        # southern boundary
+        south = xr.DataArray(tmp[0, :], dims=("x"))
+        inner_w_poles = xr.concat([south, inner], dim="y")
+
+        # apply E-W boundary condition
+        west = xr.DataArray(inner_w_poles[-1, :], dims=("y"))
+        out = xr.concat([west, inner_w_poles], dim="x")
+        depth = out
+        depth = depth[:, 1:]
```

Comment:
I don't understand why using periodicity to then exclude eastern-most point

Reply:
We need a depth field with the same dimensions as the area field.  In the case of a symmetric grid, we need to add a row and column.

Reply:
yes but you're removing said column on line 504. Also if you want to plot the depth on MOC plots (V-point)
we should use xgcm to interpolate depth from T to V point

Reply:
Ah. Good point. (no pun intended)

Reply:
how about you build the extended depth array and then each diag will interp or extract according to the grid point type used